core/lib/Cardano/Address.hs

```diff
@@ -78,6 +87,29 @@ newtype Address = Address
     } deriving stock (Generic, Show, Eq, Ord)
 instance NFData Address
 
+instance ToBech32 Address where
+    bech32 addr = bech32With (addressHrp addr) $ unAddress addr
```

Comment:
This is a little dodgy because `Address` may not be a shelley address.
It would not be good to expose this as API.
`addressHrp` is good and could be used, but only within context where the type of address is known to be Shelley.

Reply:
Any address that isn't Shelley shouldn't be bech32 anyway.

Reply:
Yes, so that's one of the reasons why `instance ToBech32 Address` is dodgy.